backend/library.c

```diff
@@ -7,22 +7,50 @@
 #include "third-party/kdtree/kdtree.c"
 
 
+// static const int CAT_STATE_CALM = 0;
+static const int CAT_STATE_HISSES = 1;
+static const int CAT_STATE_WANTS_TO_FIGHT = 2;
+
+
 static double drunk_cats_g_fight_radius = 0.0;
 static double drunk_cats_g_hiss_radius = 0.0;
 
-// static const int CAT_MOOD_CALM = 0;
-static const int CAT_MOOD_HISSES = 1;
-static const int CAT_MOOD_WANTS_TO_FIGHT = 2;
-
 
-static double *recalculate_positions(
+/**
+ * Convert cat positions in the OpenGL coordinate system to the plain flatten coordinates.
+ *
+ * ## Why *plain*?
+ *
+ * Every position `(x, y)` in the OpenGL coordinate system
+ * will be scaled using the given window size and scale.
+ *
+ * ## Why *flatten*?
```

Comment:
Do we really need these "## Why *something*" questions? I think it's better to remove them since you already have the answers below